text/src/autogluon/text/text_prediction/predictor/predictor.py

```diff
@@ -0,0 +1,419 @@
+
+    @property
+    def backend(self):
+        return self._backend
+
+    def fit(self,
+            train_data,
+            tuning_data=None,
+            time_limit=None,
+            presets=None,
+            hyperparameters=None,
+            column_types=None,
+            num_cpus=None,
+            num_gpus=None,
+            num_trials=None,
+            plot_results=None,
+            holdout_frac=None,
+            seed=0):
+        """Fit the predictor
+
+        Parameters
+        ----------
+        train_data
+            Table of the training data. It can be a pandas dataframe.
+            If str is passed, `train_data` will be loaded using the str value as the file path.
+        tuning_data
+            Another dataset containing validation data reserved for tuning processes such as early
+            stopping and hyperparameter tuning.
+            This dataset should be in the same format as `train_data`.
+            If str is passed, `tuning_data` will be loaded using the str value as the file path.
+            Note: final model returned may be fit on `tuning_data` as well as `train_data`.
+            Do not provide your evaluation test data here!
+            In particular, when `num_bag_folds` > 0 or `num_stack_levels` > 0, models will be
+            trained on both `tuning_data` and `train_data`.
+            If `tuning_data = None`, `fit()` will automatically hold out some random validation
+            examples from `train_data`.
+        time_limit
+            Approximately how long `fit()` should run for (wallclock time in seconds).
+            If not specified, `fit()` will run until the model has completed training.
+        presets : str or None, optional, default is None
+            Presets defines the pre-registered configurations. By default,
+            we will use the "medium_quality_faster_train".
+            There are some other options like
+                - best_quality: Model with the best quality but will be slower
+                                for training and inference
+                - medium_quality_faster_train: Model with the medium quality
+                - lower_quality_fast_train: Model that can be trained and inferenced very fast compared to the other two options.
+            You may try to list all presets via `autogluon.text.ag_text_presets.list_keys()`.
+        hyperparameters
+            The hyper-parameters of the fit function. This can be used to specify the
+            search space and the configuration of the network.
+        column_types
+            The provided type of the columns. It will be a dictionary that maps the column name
+            to the type of the column.
+        num_cpus
+            The number of CPUs to use for each trial
+        num_gpus
+            The number of GPUs to use for each trial
+        num_trials
+            The number of trials. By default, we will use the provided number of trials in the
```

Comment:
Docs should clarify this is the number of hyperparameter-tuning trials.
Also should clarify if this overrides `hyperparameters['tune_kwargs']['num_trials']`